llvm/tools/llvm-lsp/OptRunner.h

```diff
@@ -91,12 +91,12 @@ class OptRunner {
   llvm::Expected<SmallVector<std::pair<std::string, std::string>, 256>>
   getPassListAndDescription(const std::string PipelineText) {
     // First is Passname, Second is Pass Description.
-    auto MaybeOutErr = runShellOpt(
-        {"--print-pass-numbers", "--passes", PipelineText, "--disable-output"});
```

Comment:
Why did you remove `--disable-output` from here? Without this flag, you would have IR content that would not fit into your parsing logic for pass-list.

Reply:
Also, quick note, since we are acquiring the pass list by running opt, we no longer have the luxury of generating descriptions of the passes as we earlier used to do. Maybe changing the name to `getPassList` would be sufficient.

My earlier intention was that, since we were earlier using after-pass callbacks to generate pass-list, we had the whole module available at that point, and we can perhaps generate some summary of what the pass did by looking at the module, and make this the pass description. But after moving to running opt from shell, we have lost this ability.